binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/function/KafkaStreamsFunctionBeanPostProcessor.java

```diff
@@ -176,11 +177,15 @@ private void extractResolvableTypes(String key) {
 				ClassUtils.getDefaultClassLoader());
 		try {
 			Method[] methods = classObj.getDeclaredMethods();
-			Optional<Method> functionalBeanMethods = Arrays.stream(methods).filter(m -> m.getName().equals(key)).findFirst();
+			Optional<Method> functionalBeanMethods = KafkaStreamsBinderUtils.findTargetMethod(key, methods);
```

Comment:
Now that I see this block of code, it is identical to the code in FDC above. It might we worth exploring moving it up as a whole to the KSBU instead of just the `findTargetMethod`, I would envision something like `findFunctionByName`.  Wdyt?

Reply:
Ya, that is certainly a great idea. I would probably do that outside of this PR though.

Reply:
Issue created to track: https://github.com/spring-cloud/spring-cloud-stream/issues/2739